### DISCUSSION TITLE
Creating a Maximal Mid-Level

### DISCUSSION THREAD

**Discussion by neilotte** [CommonCoreOntology/CommonCoreOntologies#316] [Announcements]
I want to share a potential future direction for CCO and gauge interest. 

The scope of CCO is scoped not only to being a mid-level ontology, but also containing only what is going to be useful to most end users. By contrast, BFO ontologies are supposed to reflect our best science, and our best science makes many careful distinctions that most CCO-end users will not give two hoots about and which would only serve as distractions. 

For example: 

- The NCIT includes Organism, Eukaryota, Hominidae, Primate, Vertebrata, etc. and entails there is at least a hierarchical distance of 7 between the class for BFO:Object and the class Human. 
- By contrast, CCO only represents Person, Animal, and Organism and calls it a day. 

This creates obstacles for those engaging in some domain of science--where these distinctions matter--from using CCO and CCO-aligned ontologies. 

A potential solution is to create a maximal mid-level where all careful scientific distinctions may be made that are above CCO's leaf-level. The present release of CCO (with its present classes) could then be automatically derived from this maximal ontology and released as it is today. It would be understood that CCO derives its intended semantics from the maximal version, and end users who would benefit from portions of the maximal version could substitute portions of it where needed. In this way, we could continue CCO's pursuit of practicality while giving end users with particularly robust needs in some domains options with respect to their hierarchies. 

Penny for your thoughts on this: @alanruttenberg @mark-jensen @johnbeve @APCox

**Comment by alanruttenberg** [comment]
I don't think there's such a thing as maximal ontology. There's "finer" but science evolves and distinctions are added. I agree there is an issue of what to do when there is a reasonable interposing class between two existing CCO terms. What to do in that case should be a matter discussed in a the style guide. It's not obvious to me that all distinctions should be in CCO - there might be distinctions that are germane to a domain-specific ontology. Dunno, as I say should be discussed. Technically, if one has a subclass b, one can add, without problem, a subclass c, c subclass b. That doesn't affect a subclass b, which remains true because of transitivity. In addition it should be noted that queries should not depend on an instance being a *direct instance* of a class. We should perhaps be defining minimal levels of reasoning that must be applied when querying CCO-based ontologies, and ensuring subclassof works as it should would be one of those.
